src/feature/commandline.c

```diff
@@ -285,6 +285,9 @@ void mArgumentsDeinit(struct mArguments* args) {
 	args->bios = 0;
 
 	HashTableDeinit(&args->configOverrides);
+#ifdef _WIN32
+	FreeConsole();
```

Comment:
Why here, of all places?

Reply:
because it is called before returning from `main`

Reply:
The `mArguments` struct, which is what this function pertains to, and the Windows console are logically completely distinct. `AttachConsole` is called in the two main functions, SDL and Qt, and shouldn't be matched in a completely separate subsystem. Further, this function gets called in some places that aren't `main` and thus don't have a matching call to `AttachConsole`. The fact that it's called at the end of `main` is incidental. It's repeated in SDL's `main` because for most of the function that's the only thing that needs to be cleaned up. If the console also needs to be cleaned up (does it?) then you will need to modify SDL's `main` instead. Perhaps a `goto cleanup` type scenario like you can find in `perf-main.c`--which, by the way, is one of the callers of `mArgumentsDeinit` that _doesn't_ have a matching `AttachConsole` call, since it's not built as a GUI app.